python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -109,66 +93,30 @@ class AssetGraphDiffer:
-        deployment workspaces, the base deployment and the branch deployment.
-
-        We cannot make RemoteAssetGraphs directly from the workspaces because if multiple code locations
-        use the same asset key, those asset keys will override each other in the dictionaries the RemoteAssetGraph
-        creates (see from_repository_handles_and_asset_node_snaps in RemoteAssetGraph). We need to ensure
-        that we are comparing assets in the same code location and repository, so we need to make the
-        RemoteAssetGraph from an ExternalRepository to ensure that there are no duplicate asset keys
-        that could override each other.
-        """
-        check.inst_param(branch_workspace, "branch_workspace", BaseWorkspaceRequestContext)
-        check.inst_param(base_workspace, "base_workspace", BaseWorkspaceRequestContext)
-
-        branch_repo = _get_remote_repo_from_context(
-            branch_workspace, code_location_name, repository_name
-        )
-        if branch_repo is None:
-            raise DagsterInvariantViolationError(
-                f"Repository {repository_name} does not exist in code location {code_location_name} for the branch deployment."
-            )
-        base_repo = _get_remote_repo_from_context(
-            base_workspace, code_location_name, repository_name
-        )
-        return AssetGraphDiffer(
-            branch_asset_graph=branch_repo.asset_graph,
-            base_asset_graph=base_repo.asset_graph if base_repo is not None else None,
-        )
-
     def _compare_base_and_branch_assets(
         self, asset_key: "AssetKey", include_diff: bool = False
     ) -> AssetDefinitionDiffDetails:
         """Computes the diff between a branch deployment asset and the
         corresponding base deployment asset.
         """
-        if self.base_asset_graph is None:
+        if not self._base_asset_graph or not self._base_asset_graph.has(asset_key):
             # if the base asset graph is None, it is because the asset graph in the branch deployment
             # is new and doesn't exist in the base deployment. Thus all assets are new.
             return AssetDefinitionDiffDetails(change_types={AssetDefinitionChangeType.NEW})
 
-        if not self.base_asset_graph.has(asset_key):
-            return AssetDefinitionDiffDetails(change_types={AssetDefinitionChangeType.NEW})
-
-        if not self.branch_asset_graph.has(asset_key):
+        if not self._branch_asset_graph.has(asset_key):
             return AssetDefinitionDiffDetails(change_types={AssetDefinitionChangeType.REMOVED})
 
-        branch_asset = self.branch_asset_graph.get(asset_key)
-        base_asset = self.base_asset_graph.get(asset_key)
+        base_asset = self._base_asset_graph.get(asset_key).resolve_to_singular_repo_scoped_node()
+
+        branch_asset = self._branch_asset_graph.get(
+            asset_key
+        ).resolve_to_singular_repo_scoped_node()
```

Comment:
I think its at least worth leaving a comment here to highlight that edge case changes that effect the `resolve_` logic could lead to some potentially confusing results. For example I believe with this setup moving an asset to a different repo in a branch would display as no change. 

We don't really communicate the details of how an asset is defined across multiple code locations in the product across the board so some slight weirdness here is probably not worth worrying much about. Do think its worth leaving a little paper trail incase someone does hit something tho.